### PR TITLE
Add a job for nightly runs against the latest dependencies

### PR DESCRIPTION
**Pull request by hongdai**
* Note that this isn't tested as I haven't found a way to test without checkin.
* I aslo add an optional PR trigger.

### CODE REVIEW

**Comment by iamjasonp** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -184,4 +184,46 @@ Utilities.addGithubPRTrigger(prCCJob, 'Code Coverage Windows Debug', '@dotnet-bo
         Utilities.addStandardOptions(newJob)
         Utilities.addXUnitDotNETResults(newJob, 'bin/tests/**/testResults.xml')
     }
+
```
Comment (iamjasonp) [inline netci.groovy]: I think you're missing a } here

File: netci.groovy [hunk 2]
```diff
@@ -184,4 +184,46 @@ Utilities.addGithubPRTrigger(prCCJob, 'Code Coverage Windows Debug', '@dotnet-bo
...
+// **************************
+// Outerloop and Innerloop against the latest dependencies on Windows. Rolling daily for debug and release
+// **************************
+
+['Debug', 'Release'].each { configuration ->
```
Comment (iamjasonp) [inline netci.groovy]: Not Linux as well? Latest dependencies might work differently between Linux and Windows. 
Is having both Debug and Release valuable here? This is effectively just a sanity check. 

I'm thinking that having just a Linux Debug and Windows Debug is probably good enough?

File: netci.groovy [hunk 3]
```diff
@@ -184,4 +184,46 @@ Utilities.addGithubPRTrigger(prCCJob, 'Code Coverage Windows Debug', '@dotnet-bo
...
+    def configurationJobName = configuration.toLowerCase()
+    def jobName = "LatestDependencies_windows_${configurationJobName}"
```
Comment (iamjasonp) [inline netci.groovy]: by convention - lowercase jobnames, say: `latest_dependencies_${os}_${configurationJobName}`

File: netci.groovy [hunk 4]
```diff
@@ -184,4 +184,46 @@ Utilities.addGithubPRTrigger(prCCJob, 'Code Coverage Windows Debug', '@dotnet-bo
...
+
+    def newLatestDepPRJob = job(Utilities.getFullJobName(project, jobName, true)) {
+        label('windows-elevated')
+        steps {
+            batchFile("build.cmd /p:Configuration=Windows_NT_${configuration}  /p:FloatingTestRuntimeDependencies=true /p:WithCategories=\"InnerLoop;OuterLoop\"")
+        }
+    }
+
+    // Add a PR trigger
+    Utilities.addGithubPRTrigger(newLatestDepPRJob, "Outerloop and Innerloop against the latest dependencies Windows ${configuration} Build and Test", '@dotnet-bot test latest dependencies please')
```
Comment (iamjasonp) [inline netci.groovy]: This is a super long name. Consider just "Latest package dependencies ${os} ${configuration} Build and Test"

File: netci.groovy [hunk 5]
```diff
@@ -184,4 +184,46 @@ Utilities.addGithubPRTrigger(prCCJob, 'Code Coverage Windows Debug', '@dotnet-bo
...
+    Utilities.addStandardNonPRParameters(newLatestDepRollingJob)
+    Utilities.addPeriodicTrigger(newLatestDepRollingJob, '@daily')
+
+    // Create the new PR job for on demand execution.  No automatic PR trigger.
+    // Triggered with '@dotnet-bot test latest dependencies please'
+
+    def newLatestDepPRJob = job(Utilities.getFullJobName(project, jobName, true)) {
+        label('windows-elevated')
+        steps {
+            batchFile("build.cmd /p:Configuration=Windows_NT_${configuration}  /p:FloatingTestRuntimeDependencies=true /p:WithCategories=\"InnerLoop;OuterLoop\"")
```
Comment (iamjasonp) [inline netci.groovy]: Extract command to a variable

**Comment by iamjasonp** [comment]
Also, please git commit --amend your commit comment... we don't want commit comments like 

> Note that this isn't tested as I haven't found a way to test without checkin.
> I aslo add an optional PR trigger.

making it into the repo. Thanks!

**Comment by iamjasonp** [comment]
@hongdai  - as above, you can "@dotnet-bot test ci please" and inspect the results in http://dotnet-ci.cloudapp.net/job/dotnet_wcf/job/master/job/GenPRTest/

File: netci.groovy (hunk 2 above)
Comment (hongdai) [inline netci.groovy]: Run against latest has not implemented yet for Linux. There is a active issue tracking it.

File: netci.groovy (hunk 3 above)
Comment (hongdai) [inline netci.groovy]: Will make it lower case.

File: netci.groovy (hunk 5 above)
Comment (hongdai) [inline netci.groovy]: Will do.

File: netci.groovy (hunk 4 above)
Comment (hongdai) [inline netci.groovy]: Make sense. Will shorten it.

File: netci.groovy (hunk 1 above)
Comment (hongdai) [inline netci.groovy]: Good catch! Will add it.

**Comment by hongdai** [comment]
@dotnet-bot test ci please

**Comment by hongdai** [comment]
@dotnet-bot test ci please

**Comment by hongdai** [comment]
@dotnet-bot test ci please

**Comment by hongdai** [comment]
@dotnet-bot test ci please

**Comment by iamjasonp** [comment]
Once the git commit comment is amended, LGTM

**Comment by hongdai** [comment]
@dotnet-bot test ci please

**Pull request merged**